### DISCUSSION TITLE
How to move marker along line based on cursor position?

### DISCUSSION THREAD

**Discussion by ricksmit3000** [uwdata/mosaic#201] [Q&A] [answered]
I want to move a dot along `lineY` based on the cursor position, I'm able to move the ruler along the Y axis, but I'm unsure how to move the `dot` on the X axis. 

Is there maybe a function to get the Y value based on the value of X (so in this case `$point`)?

```js
vg.ruleX({ x: $point }),
vg.lineY(vg.from("stocks", { filterBy: query }), {
  x: "Date",
  y: "Open",
  stroke: "steelblue",
}),
vg.nearestX({ as: $point }),
vg.dot({ x: $point, fill: "steelblue" }),
```

**Comment by domoritz** [comment, marked as answer]
I didn't have time to build up an example but you could try to build a query that computes y based on the `$point`.

Something like

```js
vg.Query.from("stocks")
  .select({ y: "Open" })
  .where($point.predicate())
```

**Reply by ricksmit3000** [reply]
thank you! I'll give that a try as well, for now I just used the built-in tooltip provided by plot
